components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/add-step1.jsp

```diff
@@ -59,6 +59,9 @@
     boolean isAskPasswordEnabled = false;
     boolean isUserOnBoardingEnabled = false;
 
+    String GREATER_THAN_CHARACTER = ">";
```

Comment:
@somindatommy 

This is fixing only the character `>`. Is that intentional? 

Means the only character is not encoded is `>`

Reply:
@tharindu-bandara yes. Other characters such as `"`,`'`,`&` and `<` are encoded properly. Issue arises as `>` is not encoded.

Reply:
In the API docs, it says,

> The encoding of the greater-than sign (>) is not required for attributes.

I'm wondering what could be the reason behind this. 

Maybe the encoding that we do here does not meet the expectation? For instance, [`Encode.forHtmlContent`](https://www2.owasp.org/owasp-java-encoder/encoder/apidocs/org/owasp/encoder/Encode.html#forHtmlContent-java.lang.String-) does encode the character `>`.


[1] [https://www2.owasp.org/owasp-java-encoder/encoder/apidocs/org/owasp/encoder/Encode.html#forHtmlAttribute-java.lang.String-](https://www2.owasp.org/owasp-java-encoder/encoder/apidocs/org/owasp/encoder/Encode.html#forHtmlAttribute-java.lang.String-)

Reply:
In the API docs, it says the following about `Encode.forHtmlContent` [1]

> It does not escape quotation characters and is thus unsafe for use with HTML attributes. Use either forHtml or forHtmlAttribute for those methods.

Hence `Encode.forHtmlAttribute` is used. To prevent HTML injection, encoding `<` as `&lt;` is enough. Therefore encoding `>` is not mandatory.

When `>` is not encoded, it is considered as a closing tag causing rendering errors in the web page. That's why encoding is necessary. 

[1] - https://www2.owasp.org/owasp-java-encoder/encoder/apidocs/org/owasp/encoder/Encode.html#forHtmlContent-java.lang.String-

Reply:
@somindatommy Thanks for the findings and the explanation!

Then we can use the [encoding for html](https://www2.owasp.org/owasp-java-encoder/encoder/apidocs/org/owasp/encoder/Encode.html#forHtml-java.lang.String-) ? Seems like this addressed the greater than the character as well.

Reply:
Fixed: 225deac693d3998c3919586cdf7b829647e41a9c